File: php/v1/examples/BulkListAssignedTargetingOptions.php
<?php
/*
 * Copyright 2021 Google LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

// Require the base class.
require_once __DIR__ . "/../BaseExample.php";

/**
 * This example lists all of the targeting options assigned to a line item
 * across targeting types, optionally filtered by a given filter expression.
 */
class BulkListAssignedTargetingOptions extends BaseExample
{
    /**
     * (non-PHPdoc)
     * @see BaseExample::getInputParameters()
     */
    protected function getInputParameters(): array
    {
        return array(
            array(
                'name' => 'advertiser_id',
                'display' => 'Advertiser ID',
                'required' => true
            ),
            array(
                'name' => 'line_item_id',
                'display' => 'Line Item ID',
                'required' => true
            ),
            array(
                'name' => 'filter',
                'display' => 'Filter',
                'required' => false
            )
        );
    }

    /**
     * (non-PHPdoc)
     * @see BaseExample::run()
     */
    public function run()
    {
        $values = $this->formValues;
        $advertiserId = $values['advertiser_id'];
        $lineItemId = $values['line_item_id'];
        $filter = $values['filter'];

        printf(
            '<h2>Listing targeting options assigned to Line Item %s</h2>',
            $lineItemId
        );

        $response = null;
        $nextPageToken = null;

        do {
            $optParams = array(
                'filter' => $filter,
                'pageToken' => $nextPageToken
            );

            // Call the API, getting all the assigned targeting options for the
            // identified line item.
            try {
                $response = $this
                    ->service
                    ->advertisers_lineItems
                    ->bulkListLineItemAssignedTargetingOptions(
                        $advertiserId,
                        $lineItemId,
                        $optParams
                    );
            } catch (\Exception $e) {
                $this->renderError($e);
                return;
            }

            if (!empty($response->getAssignedTargetingOptions())) {
                $this->printAssignedTargetingOptions(
                    $response->getAssignedTargetingOptions()
                );
            } else {
                print '<p>No assigned targeting options returned</p>';
            }

            // Update the next page token.
            $nextPageToken = $response->getNextPageToken();
        } while (
            !empty($response->getAssignedTargetingOptions())
            && !empty($nextPageToken)
        );
    }

    /**
     * (non-PHPdoc)
     * @see BaseExample::getName()
     */
    public function getName(): string
    {
        return 'Bulk List Assigned Targeting Options';
    }

    /**
     * Prints the given found assigned targeting options.
     * @param array $options found assigned targeting options to be printed.
     */
    protected function printAssignedTargetingOptions(array $options)
    {
        foreach ($options as $option) {
            printf(
                '<p>Assigned Targeting Option %s found</p>',
                $option['name']
            );
        }
    }
}

File: java/v1/src/main/java/com/google/displayvideo/api/samples/BulkListAssignedTargetingOptions.java
// Copyright 2021 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//    https://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

package com.google.displayvideo.api.samples;

import com.beust.jcommander.Parameter;
import com.google.api.client.util.Strings;
import com.google.api.services.displayvideo.v1.DisplayVideo;
import com.google.api.services.displayvideo.v1.DisplayVideo.Advertisers.LineItems;
import com.google.api.services.displayvideo.v1.model.AssignedTargetingOption;
import com.google.api.services.displayvideo.v1.model.BulkListLineItemAssignedTargetingOptionsResponse;
import com.google.displayvideo.api.samples.utils.ArgumentNames;
import com.google.displayvideo.api.samples.utils.CodeSampleParams;

/**
 * This example lists all of the targeting options assigned to a line item across targeting types.
 */
public class BulkListAssignedTargetingOptions {

  private static class BulkListAssignedTargetingOptionsParams extends CodeSampleParams {

    @Parameter(
        names = ArgumentNames.ADVERTISER_ID,
        description =
            "The ID of the parent advertiser of the line item whose targeting is being retrieved.",
        required = true)
    public Long advertiserId;

    @Parameter(
        names = ArgumentNames.LINE_ITEM_ID,
        description = "The ID of the line item whose targeting is being retrieved.",
        required = true)
    public Long lineItemId;

    @Parameter(
        names = ArgumentNames.FILTER,
        description = "The filter expression by which to filter the list results.")
    public String filter;
  }

  public static void main(String[] args) throws Exception {
    BulkListAssignedTargetingOptionsParams params = new BulkListAssignedTargetingOptionsParams();
    if (!params.parseArguments(args)) {

      // Either pass the required parameters for this example on the command line, or insert them
      // into the code here. See the parameter class definition above for descriptions.
      params.advertiserId = Long.valueOf("INSERT_ADVERTISER_ID_HERE");
      params.lineItemId = Long.valueOf("INSERT_LINE_ITEM_ID_HERE");

      // If no filter is needed, please leave filter empty.
      params.filter = "INSERT_FILTER_HERE";
    }

    DisplayVideo service = DisplayVideoFactory.getInstance(params.clientSecretsFile);

    runExample(service, params.advertiserId, params.lineItemId, params.filter);
  }

  public static void runExample(
      DisplayVideo service, long advertiserId, long lineItemId, String filter) throws Exception {

    // Configure the bulk list request and set filter.
    LineItems.BulkListLineItemAssignedTargetingOptions request =
        service
            .advertisers()
            .lineItems()
            .bulkListLineItemAssignedTargetingOptions(advertiserId, lineItemId)
            .setFilter(filter);

    // Create the response and nextPageToken variables.
    BulkListLineItemAssignedTargetingOptionsResponse response;
    String nextPageToken = null;

    do {
      // Set page token and execute the list request.
      response = request.setPageToken(nextPageToken).execute();

      // Check if response is empty.
      if (response.isEmpty()) {
        System.out.print("Bulk list request returned no Assigned Targeting Options");
        break;
      }

      // Iterate over retrieved targeting options.
      for (AssignedTargetingOption assignedOption : response.getAssignedTargetingOptions()) {
        System.out.printf("Assigned Targeting Option %s found%n", assignedOption.getName());
      }

      // Update the next page token.
      nextPageToken = response.getNextPageToken();
    } while (!Strings.isNullOrEmpty(nextPageToken));
  }
}
